src/profile-logic/process-profile.js

```diff
@@ -1083,6 +1081,17 @@ function _processThread(
     samples,
   };
 
+  // isPrivateBrowsing and userContextId are missing in firefox 97 and
+  // earlier. Also they're missing when this thread had no extra attribute at all.
+  // Let's add them to the new thread only when present in the gecko thread.
+  if (thread.isPrivateBrowsing !== undefined) {
+    newThread.isPrivateBrowsing = thread.isPrivateBrowsing;
+  }
+
+  if (thread.userContextId !== undefined) {
+    newThread.userContextId = thread.userContextId;
+  }
```

Comment:
Is there a reason why we are only adding these if they are present in the gecko profile? I guess we can also put this to the thread directly and it would be `undefined` if they are not present. I'm not saying that we should do this though, I'm just curious why you wanted to do it this way. Also I again see `Also they're missing when this thread had no extra attribute at all.` but not sure what it means

Reply:
That's what I did initially, but then I've seen that they were serialized in all snapshots. Then I've also checked that when present but undefined, they're still in the JSON after serialization. So I decided to make it "cleaner" by not having them when not present initially. I felt like I was polluting the new thread object otherwise :-)